spec/javascripts/surveys-spec.js

```diff
@@ -90,25 +87,109 @@ describe('Surveys', function () {
       expect(GOVUK.cookie(surveys.surveySeenCookieName(defaultSurvey))).toBe('1')
     })
 
-    describe("for a 'url' survey", function () {
-      it('links to the url for a smartsurvey survey with a completion redirect query parameter', function () {
-        surveys.displaySurvey(urlSurvey)
+    describe('for a url survey', function () {
+      it('replaces the current path if the survey url contains the {{currentPath}} template parameter', function () {
+        var urlSurveyWithCurrentPath = {
+          surveyType: 'url',
+          url: 'smartsurvey.com/default?c={{currentPath}}',
+          identifier: 'url-survey'
+        }
+        surveys.displaySurvey(urlSurveyWithCurrentPath)
 
-        expect($('#take-survey').attr('href')).toContain(urlSurvey.url)
         expect($('#take-survey').attr('href')).toContain('?c=' + window.location.pathname)
       })
 
-      it('sets event handlers on the survey', function () {
-        spyOn(surveys, 'setURLSurveyEventHandlers')
+      it('does not inject the current path if the survey url does not contain the {{currentPath}} template parameter', function () {
         surveys.displaySurvey(urlSurvey)
-        expect(surveys.setURLSurveyEventHandlers).toHaveBeenCalledWith(urlSurvey)
+
+        expect($('#take-survey').attr('href')).toEqual(urlSurvey.url)
+      })
+
+      it('injects the url for a non-smartsurvey survey and adds current path as a `c` param', function () {
+        var nonSmartSurveyUrlSurvey = {
+          surveyType: 'url',
+          url: 'smartsurvey.com/default?c={{currentPath}}',
+          identifier: 'url-survey'
+        }
+        surveys.displaySurvey(nonSmartSurveyUrlSurvey)
+
+        expect($('#take-survey').attr('href')).toContain('?c=' + window.location.pathname)
       })
+    })
 
+    describe("for a 'url' survey", function () {
       it('records an event when showing the survey', function () {
         spyOn(surveys, 'trackEvent')
         surveys.displaySurvey(urlSurvey)
         expect(surveys.trackEvent).toHaveBeenCalledWith(urlSurvey.identifier, 'banner_shown', 'Banner has been shown')
       })
+
+      it('sets event handlers on the survey', function () {
+        spyOn(surveys, 'setURLSurveyEventHandlers')
+        surveys.displaySurvey(urlSurvey)
+        expect(surveys.setURLSurveyEventHandlers).toHaveBeenCalledWith(urlSurvey)
+      })
+
+      describe('without overrides for the template defaults', function () {
+        it('uses the title defined in the template data attributes', function () {
```

Comment:
We talk about template data attributes in these tests which is a hang-over from the previous implementation where the defaults did live in data attributes.  We should reword the tests to talk about template defaults or something instead.